File: javascript/easy/intersectionOfTwoArrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    let map = new Map(), res = [];
    for(let num of nums1) {
        map.set(num, true);
    }
    for(let num of nums2) {
        if(map.get(num) !== undefined) {
            res.push(num);
            map.delete(num);
        }
    }
    return res;
};

File: ruby/easy/intersection_of_two_arrays.rb
# @param {Integer[]} nums1
# @param {Integer[]} nums2
# @return {Integer[]}
def intersection(nums1, nums2)
    map, res = {}, Array.new
    nums1.each do |num|
        map[num] = true
    end
    nums2.each do |num|
        res.push(num) and map[num] = nil if !map[num].nil?
    end
    return res
end
